subsys/bluetooth/shell/bt.c

```diff
@@ -787,6 +787,101 @@ static int cmd_select(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static const char *get_conn_type_str(u8_t type)
+{
+	switch (type) {
+	case BT_CONN_TYPE_LE: return "LE";
+	case BT_CONN_TYPE_BR: return "BR/EDR";
+	case BT_CONN_TYPE_SCO: return "SCO";
+	default: return "Invalid";
+	}
+}
+
+static const char *get_conn_role_str(u8_t role)
+{
+	switch (role) {
+	case BT_CONN_ROLE_MASTER: return "master";
+	case BT_CONN_ROLE_SLAVE: return "slave";
+	default: return "Invalid";
+	}
+}
+
+static int cmd_info(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct bt_conn *conn = NULL;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	bt_addr_le_t addr;
+	struct bt_conn_info info;
+	int err;
+
+	switch (argc) {
+	case 1:
+		if (default_conn) {
+			conn = bt_conn_ref(default_conn);
+		}
+		break;
+	case 2:
+		addr.type = BT_ADDR_LE_PUBLIC;
+		err = bt_addr_from_str(argv[1], &addr.a);
+		if (err) {
+			shell_error(shell, "Invalid peer address (err %d)",
+				    err);
+			return err;
+		}
+		conn = bt_conn_lookup_addr_le(selected_id, &addr);
+		break;
+	case 3:
+		err = bt_addr_le_from_str(argv[1], argv[2], &addr);
+
+		if (err) {
+			shell_error(shell, "Invalid peer address (err %d)",
+				    err);
+			return err;
+		}
+		conn = bt_conn_lookup_addr_le(selected_id, &addr);
+		break;
+	}
+
+	if (!conn) {
+		shell_error(shell, "Not connected");
+		return -ENOEXEC;
+	}
+
+	err = bt_conn_get_info(conn, &info);
+	if (err) {
+		shell_print(ctx_shell, "Failed to get info");
+		goto done;
+	}
+
+	conn_addr_str(conn, addr_str, sizeof(addr_str));
+	shell_print(ctx_shell, "Peer address: %s", addr_str);
+
+	shell_print(ctx_shell, "Type: %s, Role: %s, Id: %u",
+		    get_conn_type_str(info.type),
+		    get_conn_role_str(info.role),
+		    info.id);
+
+	if (info.type == BT_CONN_TYPE_LE) {
+		bt_addr_le_to_str(info.le.src, addr_str, sizeof(addr_str));
+		shell_print(ctx_shell, "Initiator  address: %s", addr_str);
```

Comment:
Id use Source and Destination terminology for addresses since we may not always be the central as this seems to assume.

Reply:
I would use the GAP terms: Central and Peripheral, when printing to the shell

Reply:
> I would use the GAP terms: Central and Peripheral, when printing to the shell

Afaik they are not role depend, so src always represents the local source address and dst is always the remote destination address so we would have to check the role of the connection to do what you are suggesting which imo complicates more necessary.

Reply:
Maybe it should have been "initiator" and "responder" I'm not really sure which names are best here, but that correlates with "init_addr" and "resp_addr"
I'd like to avoid confusion here with the "dst" addr which is returned by `bt_conn_get_dst`, this is the address that should be used when re-connecting for example. It represents the resolved identity address of the peer, or the normal address if privacy is not used or not resolved yet.
The init_addr is the address that was sent on-air in the connection creation. It is used for example by SMP and will not change during the connection.

Reply:
Under what situation does an application need to know the init_addr and resp_addr? To me it should like this should be handled by stack, in fact we already have code handling similar situation in bt_conn_addr_le_cmp.

Reply:
Id stick with Source/Local and Destination/Remote addresses then since get info is doing the right thing when assigning the addresses by checking the connection role, the only thing that perhaps is missing is to use identity addresses if there is one so that would be consistent with get_dst, though the same can be said for local address as well where the address depend on the id used for connecting.

Reply:
> Id stick with Source/Local and Destination/Remote addresses then since get info is doing the right thing when assigning the addresses by checking the connection role

You are right, messed that up, 

>  the only thing that perhaps is missing is to use identity addresses if there is one so that would be consistent with get_dst

In that case we should always return dst.
Not sure what is the best thing to do here.

1. Info was meant to provide the source and destination, in which case we should have returned `&conn->le.dst` and `&bt_dev.id_addr[conn->id]`
2. Info is correct, but documentation is not clear that it is the device addresses used on-air, improve documentation?. Also there is no way to get out the source address, maybe add `bt_conn_get_src`?
3. Add all addresses in get_info? src/dst and resp/init

Ideas?

Reply:
3. might be the way to go, it seems we have the role already exposed so the application can make sense of responder and initiator addresses and the shell can be a good source to how to use them properly. As for the bt_conn_get_src that might be a good idea as well, either that or bt_conn_get_id.